File: z-clients/python/diom/apis/admin_metrics.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    GetMetricsOut,
)


class AdminMetricsAsync(ApiBase):
    async def get(
        self,
    ) -> GetMetricsOut:
        """Dump the current metrics (which would otherwise be sent to the OTLP metrics receiver)"""

        return await self._request_asyncio(
            method="get",
            path="/api/v1.admin.metrics.get",
            response_type=GetMetricsOut,
        )


class AdminMetrics(ApiBase):
    def get(
        self,
    ) -> GetMetricsOut:
        """Dump the current metrics (which would otherwise be sent to the OTLP metrics receiver)"""

        return self._request_sync(
            method="get",
            path="/api/v1.admin.metrics.get",
            response_type=GetMetricsOut,
        )

File: z-clients/java/src/main/java/com/svix/diom/apis/AdminMetrics.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.GetMetricsOut;

public class AdminMetrics {
    private final HttpClient client;

    public AdminMetrics(HttpClient client) {
        this.client = client;
    }

    /** Dump the current metrics (which would otherwise be sent to the OTLP metrics receiver) */
    public GetMetricsOut get(
        
    ) throws DiomException {

        return this.client.executeRequest(
            "GET",
            "/api/v1.admin.metrics.get",
            null,
            null,
            GetMetricsOut.class
        );
    }
}